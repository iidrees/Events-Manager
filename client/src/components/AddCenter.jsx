
import React from 'react';
import { connect } from 'react-redux';
import axios from 'axios';
import { Redirect } from 'react-router-dom'

import NavBarMain from './NavBarMain.jsx';
import Footer from './footer.jsx';
import { addCenter, imageUpload } from '../actions/addCentersAction';
import { centerDetails } from '../actions/centerDetailsAction';


/**
 * 
 * @param {event} e {e}
 * @class AddCenter
 * @extends {React.Component}
 */
class AddCenter extends React.Component {

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  onSubmit = (e) => {
    e.preventDefault();
    let centerData = this.state;
    const { dispatch } = this.props;
    if (!centerData.imgFile) {
      return dispatch(addCenter(centerData))
    }
    return dispatch(imageUpload(centerData));
  }

/**
 * @param {e} e {event}
 * @returns {any} any
 * @memberof AddCenter
 */
onImageChange = (e) => {
    e.preventDefault();

    let reader = new FileReader();
    let imgFile = e.target.files[0];

    if (imgFile.type === 'image/jpeg' || imgFile.type === 'image/png') {
      reader.onloadend = () => {
        this.setState({
          imgFile: imgFile,
          imgUrl: reader.result
        });
      }
      reader.readAsDataURL(imgFile);
    } else {
      alert('Please upload an image with the .jpeg or .png file format')
    }
  
  }


/**
 * 
 * 
 * @returns {String} return the HTML markup to be rendered 
 * @memberof AddCenter
 */
render () {

    const { createCenter, user } =  this.props;
    console.log('add center component >>>>', createCenter, ' and User reducer', user)
    return (
      <div>
        <div>
          {
            (!user.authenticated) &&
            <Redirect to='/signin' push />
          }
        </div>
        <NavBarMain />
        <div className="container">
          <div className="row">
            <div className="container" id="add-center-header">
              <div className="row">
                <div className="col-sm-12">              
                  <h1 className="text-center head-1" id="admin-add-head">Add New Event Centers Below </h1>
                  <p className="head-para text-center">
                  
                  </p>
                  <hr /> 
                </div>
              </div>
            </div>
      
           {(createCenter.status === 'Success' ) && <div className="alert alert-success" role="alert">
												<button type="button" className="close" data-dismiss="alert" aria-label="Close">
													<span aria-hidden="true">&times;</span>
												</button>
                        <strong>{createCenter.message}.</strong></div>}
                        {(createCenter.status === 'Unsuccessful' ) && <div className="alert alert-danger" role="alert">
												<button type="button" className="close" data-dismiss="alert" aria-label="Close">
													<span aria-hidden="true">&times;</span>
												</button>
												<strong>{createCenter.message}</strong><span> </span>
												<strong>{createCenter.error}.</strong></div>}
            <div className="container">{/* <!-- Start container for Add Form --> */}
              <div className="row">
                <div className="col-md-6">
                  <form method="POST" className="form form-center" onSubmit={this.onSubmit} role="form" id="admin-form"  action="#">{/* <!-- ADD FORM --> */}
                    <div className="form-group">
                      <label htmlFor="add-center" className=" home-para">Name of Center:</label>
                      <input className="form-control" onChange={this.onChange} name="name" type="text" placeholder="Enter Name of Center" id="example-text-input" />
                    </div>
                    <div className="form-group">
                      <label htmlFor="add-center" className=" home-para">Location:</label>
                      <input className="form-control" type="text" placeholder=" Enter Location" id="example-text-input" name="location" onChange={this.onChange}  />
                    </div>                
                    <div className="form-group">
                      <label htmlFor="add-center" className=" home-para">Address:</label>
                      <input className="form-control" type="text" placeholder="Enter Address" id="example-text-input" name="address" onChange={this.onChange}/>
                    </div>
                    <div className="form-group">
                      <label htmlFor="add-center" className=" home-para">Capacity:</label>
                      <input className="form-control" type="text" placeholder="Enter Capacity" id="example-text-input" name="capacity" onChange={this.onChange} />
                    </div>
                    <div className="form-group">
                      <label htmlFor="add-center" className=" home-para">Owner:</label>
                      <input className="form-control" type="text" placeholder="Enter Owner" id="example-text-input" name="owner" onChange={this.onChange} />
                    </div>
                    <div className="form-group">
                      <label htmlFor="event-details" className=" home-para"> Description:</label>
                      <textarea className="form-control" id="eventTextarea" rows="8" placeholder="brief details about the event" name="description" onChange={this.onChange} ></textarea>
                    </div>
                    <div className="form-group">
                      <label htmlFor="add-center" className=" home-para">Upload an Image of event center below:</label>
                      <input type="file" className="form-control-file" onChange={this.onImageChange} id="input-file" name="images" aria-describedby="fileHelp" />
                    </div>  
                    <button type="submit" className="btn btn-primary btn-sm" id="save-event">Save and create Event Center<span><i className="fa fa-paper-plane" aria-hidden="true"></i></span></button>
                  </form>          
                </div>{/* <!-- End Container ADD EVENTS FORM --> */}
              </div>
            </div>
          </div>
        </div>
      </div>
    )

  }
}


const mapDispatchToProps = (dispatch) => {
  return {
    dispatch: (action) => dispatch(action)
  }
};
const mapStateToProps = (state) => {
  return {
    createCenter: state.addCenterReducer,
    user: state.userReducer
    
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(AddCenter);