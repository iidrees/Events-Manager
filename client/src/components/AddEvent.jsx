import React from 'react';
import { connect } from 'react-redux';
import { Redirect, withRouter } from 'react-router-dom';
import axios from 'axios';
import jwt from 'jsonwebtoken';
import toastr from 'toastr';
import validator from 'validator';
import { Wave, Third } from 'react-preloading-component';

import NavBarMain from './NavBarMain.jsx';
import Footer from './Footer.jsx';
import { addEvent, imageUpload } from '../actions/addEvent';
import getCenters from '../actions/getCenters';
import { history } from '../routes';
import AddEventComponent from './EventsComponents/AddEventComponent.jsx';
import AddEventHeaderComponent from './EventsComponents/AddEventHeaderComponent.jsx';

/**
 * @param {event} event {events}
 * @class Addevents
 * @extends { React.Component }
 */
class AddEvent extends React.Component {
  /**
   * @returns {void}
   * @param {event} event -
   * @memberof AddEvent
   */
  onChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  /**
   * @returns {void}
   * @param {event} event -
   * @memberof AddEvent
   */
  onSubmit = event => {
    event.preventDefault();
    let eventData = this.state;

    const { dispatch } = this.props;
    if (
      eventData.imgFile === null ||
      eventData.imgFile === undefined ||
      !eventData.imgFile
    ) {
      return dispatch(addEvent(eventData, eventData.center));
    }
    return dispatch(imageUpload(eventData, eventData.center));
  };

  /**
   * @returns {void}
   * @param {event} event -
   * @memberof AddEvent
   */
  onImageChange = event => {
    event.preventDefault();

    let reader = new FileReader();
    let imgFile = event.target.files[0];

    try {
      if (imgFile.type === 'image/jpeg' || imgFile.type === 'image/png') {
        reader.onloadend = () => {
          this.setState({
            imgFile: imgFile,
            imgUrl: reader.result
          });
        };
        reader.readAsDataURL(imgFile);
      } else {
        alert('Please upload an image with the .jpeg or .png file format');
      }
    } catch (error) {
      alert('Please upload an image with the .jpeg or .png file format');
    }
  };

  /**
   * @returns {JSON} An array of centers
   * @memberof Addevents
   */
  componentDidMount() {
    const { dispatch } = this.props;
    return dispatch(getCenters());
  }

  /**
   * @returns {any} -
   * @memberof Addevents
   */
  render() {
    const { user, centers, event } = this.props;
    let userId, token, decoded;
    try {
      token = localStorage.getItem('x-access-token');

      userId = jwt.decode(token).id;

      if (jwt.decode(token).admin) {
        return <Redirect to="/getCenters" push />;
      }
    } catch (error) {
      decoded = null;
    }

    if (event.status === 'Unsuccessful') {
      toastr.options.preventDuplicates = true;
      toastr.options.positionClass = 'toast-top-left';
      toastr.error(`${event.message}`);
      event.status = '';
    }
    if (event.status === 'Success') {
      toastr.options.preventDuplicates = true;
      toastr.options.positionClass = 'toast-top-left';
      toastr.success(`${event.message}`);
      event.status = '';
      return <Redirect to="/myevents" push />;
    }
    if (event.isLoading) {
      event.isLoading = false;
      return (
        <div style={{ paddingTop: '350px' }}>
          <Wave />
        </div>
      );
    }
    return (
      <div>
        <div className="container" id="add-events">
          <div className="row">
            <AddEventHeaderComponent />
            <div className="container">
              {/* <!-- Start container for Add Form --> */}
              <div className="row">
                <AddEventComponent
                  {...centers}
                  onChange={this.onChange}
                  onSubmit={this.onSubmit}
                  onImageChange={this.onImageChange}
                />
              </div>
            </div>
          </div>
          {/* <!-- End container for the Head section --> */}
        </div>
      </div>
    );
  }
}

/**
 * Maps dispatch to props
 * @param {any} dispatch -
 * @returns {void}
 */
const mapDispatchToProps = dispatch => {
  return {
    dispatch: action => dispatch(action)
  };
};

/**
 * Maps state to props and the store is exposed to the component
 *
 * @param {any} state -
 * @returns {void}
 */
const mapStateToProps = state => {
  return {
    event: state.addEventReducer,
    user: state.userReducer,
    centers: state.centerReducer,
    token: state.userTokenReducer
  };
};
export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(AddEvent)
);
