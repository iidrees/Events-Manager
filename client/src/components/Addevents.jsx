import React from 'react';
import { connect } from 'react-redux';
import axios from 'axios';

import NavBarMain from './NavBarMain.jsx';
import Footer from './footer.jsx';
import { addEvent } from '../actions/addEventActions';
import  getCenters  from '../actions/getCentersAction';


/**
 * 
 * @class Addevents
 * @extends { React.Component }
 */
class Addevents extends React.Component {



  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  onSubmit = (e) => {
		e.preventDefault();
    let eventData = this.state;
   
		const { dispatch } = this.props;
		return dispatch(addEvent(eventData, eventData.center));
  }
  /* eslint-disable */
  componentDidMount() {
    const { dispatch } = this.props;
    return dispatch(getCenters());
  }

  /**
   * 
   * 
   * @returns {any} - 
   * @memberof Addevents
   */
  render() {
    
    const { centers } = this.props;
  return (
    <div>
    <NavBarMain />
    <div className="container">
    <div className="row">
      <div className="container" id="add-event-header">
        <div className="row">
          <div className="col-sm-12">              
            <h1 className="text-center head-1">Add your events </h1>
            <p className="head-para text-center">
              Are you in need of a location to host your event? 
              If yes, then why not host your events using one of 
              our numerous event centers by creating an event below?
            </p>
          </div>
        </div>
      </div>
    
      
      <div className="container">{/* <!-- Start container for Add Form --> */}
        <div className="row">
          <div className="col-sm-12">
            <form method="POST" className="form-horizontal" onSubmit={this.onSubmit} role="form"  action="#">{/* <!-- ADD FORM --> */}
              <div className="form-group">
                <label htmlFor="add-event" className=" home-para">Name of Event:</label>
                <input className="form-control" onChange={this.onChange}  name="title" id="form-event1" type="text" placeholder=""  />
              </div>
              <div className="form-group">
                <label htmlFor="add-event" className=" home-para">Date of event:</label>
                <input className="form-control" onChange={this.onChange}  name="date" type="text" id="form-event2"placeholder="YY/MM/DD"  />
              </div>              
              <div className="form-group">
                <label htmlFor="add-event" className=" home-para">Time:</label>
                <input className="form-control" onChange={this.onChange}  name="time" type="text" id="form-event7"placeholder="hh:mm AM/PM"  />
              </div>
              <div className="form-group">
                <label htmlFor="event-centers" className=" home-para">Center:</label>
                <select className="form-control" id="event-center1" onChange={this.onChange}  name="center">
                  <option className="home-para">Choose an event center</option>
                  {(centers.map((center) => {
                    
                    return (<option key={center.id} value={center.id} className="home-para">{center.name}</option>)
                  }))}
                  
                </select>
              </div>
              <div className="form-group">
                <label htmlFor="event-details"  className=" home-para">Events description:</label>
                <textarea className="form-control" onChange={this.onChange} id="form-event4" rows="8" name="description" placeholder="brief details about the event"></textarea>
              </div>
              <div className="form-group">
                <label htmlFor="event-centers" className=" home-para">Event type:</label>                
                <input className="form-control" onChange={this.onChange}  name="type" type="text" id="event-center2" placeholder="What kind of event is it, 'public' or 'private'?"  />          
              </div>
              <div className="form-group">
                <label htmlFor="add-event" className=" home-para">Upload an Image of your event below:</label>
                <input type="file" className="form-control-file" onChange={this.onChange}  aria-describedby="fileHelp" /> 
              </div>  
          
              <button type="submit" className="btn btn-primary btn-sm" id="save-event">Save and create event  <span><i className="fa fa-paper-plane" aria-hidden="true"></i></span></button>
            </form>          
          </div>{/* <!-- End Container ADD EVENTS FORM --> */}
        </div>
      </div>
  </div>{/* <!-- End container for the Head section --> */}
  </div>
  <Footer />
  </div>
  );
}
}


const mapDispatchToProps = (dispatch) => {
  return {
    dispatch: (action) => dispatch(action)
  }
};
const mapStateToProps = (state) => {
  return {
    addevents: state.eventReducer,
    status: state.userReducer,
    centers: state.centerReducer
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(Addevents);