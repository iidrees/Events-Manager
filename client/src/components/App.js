import React from 'react';
import { Route, HashRouter, Switch, Link, BrowserRouter, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import NavigationBar  from './NavigationBar.js';
import Main from './Main.jsx'
import Getevents from './Getevents.jsx'
import Footer from './footer.jsx';
import  './scss/style.scss';
import  './css/style.css';

/**
 * This is the App component that helps
 * abstract out the implementation of the sign-up component while using the react-router-dom
 * @class App
 * @extends {React.Component}
 */
class App extends React.Component {
  render() {
    const { userSignupRequest } = this.props
    return (
      <div className="container">
        <NavigationBar />
        <BrowserRouter>
          <Switch>            
            <Route exact path='/users' component={Main} />
            <Route exact path='/getEvents' component={Getevents} />
          </Switch>
        </BrowserRouter>
        <Footer />
      </div>
    );
  }
}
export default App;