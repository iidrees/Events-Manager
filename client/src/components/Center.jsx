import React from 'react';
import { connect } from 'react-redux';
import { Link, Redirect, withRouter } from 'react-router-dom';
import Pagination from 'rc-pagination';
import jwt from 'jsonwebtoken';
import toastr from 'toastr';

import Footer from './Footer.jsx';
import getCenters from '../actions/getCentersAction';
import NavBarMain from './NavBarMain.jsx';
import { history } from '../routes';
import GetCentersComponent from './CentersComponents/GetCentersComponent.jsx';
import GetCentersHeaderComponent from './CentersComponents/GetCentersHeaderComponent.jsx';
/**
 * @class Center
 * @extends {React.Component}
 */
class Center extends React.Component {
  /**
   * Creates an instance of Center.
   * @param {any} props -
   * @memberof Center
   */
  constructor(props) {
    super(props);
    this.state = {
      currentPage: 1,
      itemsPerPage: 10,
      userName: '',
      isAdmin: false
    };
    this.onChange = this.onChange.bind(this);
  }

  /**
   * @returns {*} function
   * @memberof Center
   */
  componentDidMount() {
    const { dispatch, currentPage } = this.props;
    return dispatch(getCenters(currentPage));
  }
  /**
   *
   * @returns {any} -
   * @param {any} page -
   * @memberof Center
   */
  onChange = page => {
    this.setState(
      {
        currentPage: page
      },
      () => {
        const { dispatch } = this.props;
        return dispatch(getCenters(this.state.currentPage));
      }
    );
  };

  /**
   * @returns {*} any
   * @memberof Center
   */
  render() {
    const { centers, user, currentPage, itemsPerPage } = this.props;
    let userId, token, decoded;
    try {
      token = localStorage.getItem('x-access-token');
      userId = jwt.decode(token).id;
    } catch (error) {
      decoded = null;
    }
    if (centers.status === 'Unsuccessful') {
      toastr.options.preventDuplicates = true;
      toastr.options.positionClass = 'toast-top-left';
      toastr.error(
        'No Event Centers Available at the Moment, check back later'
      );
      centers.status = '';
    }
    if (centers.status === 'Success') {
      toastr.options.preventDuplicates = true;
      toastr.options.positionClass = 'toast-top-left';
      toastr.success('These are the event centers available');
      centers.status = '';
    }
    return (
      <div>
        <div className="container">
          <div>
            <GetCentersHeaderComponent centers={centers} />
            <GetCentersComponent centers={centers} userId={userId} />
          </div>
        </div>
        <div className="col-sm-12 col-md-6 col-lg-4 paginator">
          {
            <Pagination
              onChange={this.onChange}
              current={this.state.currentPage}
              className="pagination"
              total={centers.count}
              locale={{ items_per_page: 'items' }}
            />
          }
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    dispatch: action => dispatch(action)
  };
};
const mapStateToProps = state => {
  return {
    centers: state.centerReducer,
    user: state.userReducer
  };
};
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Center));
