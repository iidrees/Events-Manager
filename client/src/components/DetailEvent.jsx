import React from 'react';
import { connect } from 'react-redux';
import { Link, Redirect, withRouter } from 'react-router-dom';
import jwt from 'jsonwebtoken';
import toastr from 'toastr';
import _ from 'lodash';

import NavBarMain from './NavBarMain.jsx';
import Footer from './Footer.jsx';
import { detailEvent } from '../actions/eventAction';
import { deleteEvent } from '../actions/deleteEventAction';
import { history } from '../routes';
import DetailsComponent from './EventsComponents/DetailsComponent.jsx';
import DetailsHeaderComponent from './EventsComponents/DetailsHeaderComponent.jsx';
/**
 *
 *
 * @className GetEvents
 * @extends {React.Component}
 */
class DetailEvent extends React.Component {
  /**
   * A method that lets a user get their events
   * @memberof DetailEvent
   * @returns {any} dispatched actions
   */
  componentDidMount() {
    const { dispatch } = this.props;
    return dispatch(detailEvent(this.props.match.params.id));
  }

  /**
   * @param {events} events is the event
   * @returns {action} returns a dispatched action
   * @memberof DetailEvent
   */
  onDelete = events => {
    events.preventDefault();
    let { event } = this.props;
    const { dispatch } = this.props;
    toastr.options.preventDuplicates = true;
    toastr.options.positionClass = 'toast-top-left';
    toastr.success('Event Deleted Successfully');
    dispatch(deleteEvent(event.id));
    return history.push('/myevents');
  };

  /**
   *
   *
   * @returns {*} JSX
   * @memberof DetailEvent
   */
  render() {
    const { event, user } = this.props;

    let userId, token, decoded;

    try {
      token = localStorage.getItem('x-access-token');
      userId = jwt.decode(token).id;
      if (jwt.decode(token).admin) {
        return <Redirect to="/getCenters" push />;
      }
    } catch (error) {
      decoded = null;
    }
    if (event.status === 'Unsuccessful') {
      toastr.options.preventDuplicates = true;
      toastr.options.positionClass = 'toast-top-left';
      toastr.error(`${event.message} Event not available`);
      event.status = '';
    }

    return (
      <div>
        {_.isEmpty(event.message) ? (
          <Redirect to="/myevents" push />
        ) : (
          <div>
            {' '}
            <div className="container" id="myevent">
              <div className="row">
                <div className="container">
                  {/* <!-- START BODY-HEADER WITH SEARCH FORM --> */}
                  <div className="row">
                    <DetailsHeaderComponent event={event} />
                  </div>
                  <DetailsComponent event={event} onDelete={this.onDelete} />
                </div>
                {/* <!-- END BODY-HEADER WITH SEARCH FORM --> */}
              </div>
            </div>
            <div id="card-body1" />
            <Footer />
          </div>
        )}
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    dispatch: action => dispatch(action)
  };
};
const mapStateToProps = state => {
  return {
    event: state.detailsEventReducer,
    user: state.userReducer,
    token: state.userTokenReducer
  };
};
export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(DetailEvent)
);
