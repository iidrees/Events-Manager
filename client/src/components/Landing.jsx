/*  eslint-disable */
import React from 'react';
import { Link } from 'react-router-dom';

import Footer from './footer.jsx';
/**
 *
 *
 * @export
 * @class LandingPage
 * @extends {React.Component}
 */
export default class LandingPage extends React.Component {
  /**
   *
   *
   * @returns
   * @memberof LandingPage
   */
  render() {
    return (
      <div id="landing-background">
        <div className="container" id="landing-page">
          <div className="row">
            <div className="col-sm-6  col-md-12 col-lg-6" id="heading">
              <h1 className="event-header">Events Manager</h1>{' '}
              {/* <!-- Intro Header --> */}
            </div>
          </div>
        </div>
        <hr />
        <div className="container" id="body-landing">
          <div className="row">
            <div className="col-sm-8 col-md-8 col-lg-12">
              <p id="para-landing">
                For All Event Planners, Party-Goers, And Owambe Finders
              </p>
            </div>
          </div>
        </div>
        <hr />

       
      </div>
    );
  }
}
