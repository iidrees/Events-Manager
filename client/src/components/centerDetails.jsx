import React from 'react';
import { connect } from 'react-redux';
import { Link, Redirect } from 'react-router-dom';

import NavBarMain from './NavBarMain.jsx';
import Footer from './footer.jsx';
import { deleteCenter } from '../actions/deleteCenterAction';
import {centerDetails } from '../actions/centerDetailsAction';

/**
 * 
 * 
 * @class CenterDetails
 * @extends {React.Component}
 */
class CenterDetails extends React.Component {
/**
 * 
 * 
 * @returns {JSON} JSON
 * @memberof CenterDetails
 */
componentWillMount() {
    const { dispatch } = this.props;
    console.log('This is details dispatch >>>>')
    return dispatch(centerDetails(this.props.match.params.id));
  }
/**
 * 
 * @returns {null} null
 * @param {event} event -
 * @memberof CenterDetails
 */
onDelete = (event) => {
    event.preventDefault();
    let { center } = this.props;
    const { dispatch } = this.props;
    return dispatch(deleteCenter(center.id));
  }
/**
 * 
 * 
 * @returns {JSX} JSX
 * @memberof CenterDetails
 */
render () {

    const { center, user } = this.props;

    return (
      <div>
        <div>
          {
            (!user.authenticated) &&
            <Redirect to='/signin' push />
          }
        </div>
      <div className="container">
        <div className="row">
          <div className="container" id="center-details-header">{/* START PAGE HEADER */}
            <div className="row">
              <div className="col-sm-12">              
                  <h1 className="text-center head-1">Event center details </h1>
                    <p className="head-para text-center">
                    The {center.name}
                    </p>
              </div>
            </div>
            <hr/>
          </div>{/* END PAGE-HEADER */}

          <div className="container" id="center-details-slider"> {/* <!-- START OF EVENT CENTER DETAILS --> */}
            <div className="row">
              <div className="col-md-6">{/* <!--  START OF SLIDER COLUMN --> */}
                  <div id="carousel-landing" className="carousel slide" data-ride="carousel">{/* <!-- START TAG FOR CAROUSEL SLIDER   --> */}
                    <ol className="carousel-indicators">
                      <li data-target="#carousel-landing" data-slide-to="0" className="active"></li>
                      <li data-target="#carousel-landing" data-slide-to="1"></li>
                      <li data-target="#carousel-landing" data-slide-to="2"></li>
                    </ol>
                    <div className="carousel-inner" role="listbox">
                      <div className="carousel-item active">
                        <img id="img-landing1" className="d-block img-fluid" src="https://static.pexels.com/photos/3865/lights-night-firework-new-year-s-eve.jpg" style={{width: '100%', height: '65%'}} alt="First slide"/>
                      </div>
                      <div className="carousel-item">
                        <img id="img-landing2" className="d-block img-fluid" src="https://static.pexels.com/photos/274192/pexels-photo-274192.jpeg"  style={{width: '100%', height: '65%'}} alt="Second slide"/>
                      </div>
                      <div className="carousel-item">
                        <img id="img-landing3" className="d-block img-fluid" src="https://static.pexels.com/photos/169190/pexels-photo-169190.jpeg"  style={{width: '100%', height: '65%'}} alt="Third slide"/>
                      </div>
                      <div className="carousel-item">
                          <img id="img-landing4" className="d-block img-fluid" src="https://static.pexels.com/photos/169211/pexels-photo-169211.jpeg"  style={{width: '100%', height: '65%'}} alt="Third slide"/>
                      </div>
                    </div>
                    <a className="carousel-control-prev" href="#carousel-landing" role="button" data-slide="prev">
                      <span className="carousel-control-prev-icon" aria-hidden="true"></span>
                      <span className="sr-only">Previous</span>
                    </a>
                    <a className="carousel-control-next" href="#carousel-landing" role="button" data-slide="next">
                      <span className="carousel-control-next-icon" aria-hidden="true"></span>
                      <span className="sr-only">Next</span>
                    </a>
                </div>{/* <!--  END Carousel slider Tag --> */}
              </div>{/* <!-- END OF SLIDER COLUMN --> */}
              
              <div className="col-md-6">
                <p className="center-details-para">
                  <span className="center-details-span">Center:</span>
                  The {center.name}
                </p>
                <p className="center-details-para">
                  <span className="center-details-span">Description:</span> <br/>
                  {center.description}
                </p>
                <p className="center-details-para">
                  <span className="center-details-span">Location:</span>
                  {center.location}
                </p>
                <p className="center-details-para">
                  <span className="center-details-span">Capacity:</span>
                  {center.capacity}
                </p>
                <p className="center-details-para">
                  <span className="center-details-span" >Owner:</span>
                  {center.owner}
                </p>
                <p className="center-details-para">
                </p>
                <Link className="btn btn-primary" to={`/editcenter/${center.id}`} role="button">Edit Center Details</Link>
                {/* <button className="btn btn-danger" onClick={this.onDelete}  role="button">DELETE CENTER</button> */}
                <button type="button" className="btn btn-danger" data-toggle="modal" data-target="#deleteCenter">
                    DELETE
                  </button>
                {/* <!-- Modal --> */}
                <div className="modal fade" id="deleteCenter" tabIndex="-1" role="dialog" aria-labelledby="deleteCenterLabel" aria-hidden="true">
                    <div className="modal-dialog" role="document">
                      <div className="modal-content">
                        <div className="modal-header">
                          <h5 className="modal-title" id="deleteCenterLabel">ARE YOU SURE YOU WANT TO DELETE THIS CENTER?</h5>
                          <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                            <span aria-hidden="true">&times;</span>
                          </button>
                        </div>
                        <div className="modal-body">
                        The {center.name}
                        </div>
                        <div className="modal-footer">
                          <button type="button" className="btn btn-primary" data-dismiss="modal">No</button>
                          <button className="btn btn-warning" onClick={this.onDelete} type="button">Delete Center</button>
                          {/* <button type="button" className="btn btn-primary">Save changes</button> */}
                        </div>
                      </div>
                    </div>
                  </div>
              </div>
            </div>
            <hr/>
          </div>{/* <!-- END OF EVENT CENTER DETAILS --> */}

          <div className="container" id="event-holding-header">
            <div className="row">
              <div className="col-sm-12">              
                <h4 className="text-center" id="head-holding">
                  Events At This Event Center
                </h4>
              </div>
            </div>
            <hr/>
          </div>

          <div className="container" id="event-holding">
            <div className="row">
           {/* {center.events.map((event) => {
              return (

              <div className="col-sm-4" key={event.id}>
              <div className="card-deck cont-body" id="card-body">
                <div className="card" id="card1" style={{height: "10rem"}}>
                  <img className="card-img-top" src="https://static.pexels.com/photos/122250/pexels-photo-122250.jpeg" alt="Card image cap"/>
                  <div className="card-block" style={{borderBottom: "solid grey 0.5px"}}>
                    <h5 className="card-title even-font">
                      <span >Event:</span>
                      {event.title}
                    </h5>
                    <p className="card-title date-font">
                        <span>DATE:</span>
                      {event.date}</p>
                    <p className="card-text">
                        <span>tags:</span>
                      <small className="text-muted">{event.type}</small></p>
                  </div>              
                </div>
              </div>
            </div>
              )
            })}  */}

            </div>
          </div>
          
        </div>
      </div>
      </div>
    )
  }

}

const mapDispatchToProps = (dispatch) => {
  return {
    dispatch: (action) => dispatch(action)
  }
};
const mapStateToProps = (state) => {
  return {
    center: state.centerDetailsReducer,
    user: state.userReducer
    
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CenterDetails);