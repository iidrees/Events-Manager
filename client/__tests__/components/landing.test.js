import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount, render } from 'enzyme';

import sinon from 'sinon';

import { LandingPage } from '../../src/components/Landing';

describe('Landing component', () => {
  it('should render without throwing an error', done => {
    expect(shallow(<LandingPage />).length).toEqual(1);
    done();
  });

  it('should render while storing token in localstorage', done => {
    const wrapper = shallow(<LandingPage />);
    const action = wrapper.instance();
    const componentDidMount = jest.spyOn(
      wrapper.instance(),
      'componentDidMount'
    );
    action.componentDidMount();

    expect(componentDidMount).toBeCalled();
    done();
  });
});
