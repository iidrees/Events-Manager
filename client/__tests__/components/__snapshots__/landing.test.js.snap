// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing component should render while storing token in localstorage 1`] = `
[MockFunction] {
  "calls": Array [
    Array [],
  ],
}
`;
